src/infra/marketplace/marketplaceCmdRepo.go

```diff
@@ -305,3 +306,101 @@ func (repo *MarketplaceCmdRepo) InstallItem(
 
 	return repo.persistInstalledItem(catalogItem, installDir, installUuid)
 }
+
+func (repo *MarketplaceCmdRepo) getInstalledItemUnusedServices(
+	installedItemServices []valueObject.ServiceName,
+) ([]valueObject.ServiceName, error) {
+	unusedServiceNames := []valueObject.ServiceName{}
+
+	installedItems, err := repo.marketplaceQueryRepo.GetInstalledItems()
+	if err != nil {
+		return unusedServiceNames, errors.New("InstalledItemsNotFound")
+	}
+
+	if len(installedItems) == 0 {
+		unusedServiceNames = installedItemServices
+		return unusedServiceNames, nil
+	}
+
+	servicesInUse := []valueObject.ServiceName{}
+	for _, installedItem := range installedItems {
+		servicesInUse = slices.Concat(
+			servicesInUse,
+			installedItem.ServiceNames,
+		)
+	}
+
+	for _, serviceName := range installedItemServices {
+		if slices.Contains(servicesInUse, serviceName) {
+			continue
+		}
+
+		unusedServiceNames = append(unusedServiceNames, serviceName)
+	}
+
+	return unusedServiceNames, nil
+}
+
+func (repo *MarketplaceCmdRepo) uninstallUnusedServices(
+	serviceNames []valueObject.ServiceName,
```

Comment:
Mais um problema que daria pra ser detectado lendo o código em voz alta e explicando para um leigo. O seu código não é o que você explica. Veja só.

Sua função chama-se "uninstallUnusedServices" e pede um parâmetro chamado "serviceNames []valueObject.ServiceName", logo, assumo que esse serviceNames são os serviços a serem desinstalados.

Daí na linha seguinte, você tem uma variável chamada unusedServiceNames. What? O parâmetro da função já não era os unuseds? Qual a diferença aqui?

Dentro da getInstalledItemUnusedServices() o seu parâmetro "serviceNames" agora se chama "installedItemServices". Wait, what? Não era os serviços a serem desinstalados? Por que está sendo passado para uma função que pega os serviços que não estão em uso pelos serviços a lista de serviços que estão em uso e que em teoria era na vdd a lista de serviços não usados? Caos total no raciocínio, código pra dentro.

Beleza, seguimos na linha 315 e pegamos os itens instalados. Ué, se pegamos os itens instalados, já não vamos pegar os serviços usado inclusive pelo item sendo desinstalado? Por que passaríamos algo pra essa função então se ela não precisa de nada?

Ok, linha 320 se não tiver item instalado, o retorno é a lista de serviços instalados que passamos pra ele. Wait, what? Como não haveria nada instalado se a gente tá removendo um item? O item não existe mais?

Volto a ler o método principal, a sua "uninstallUnusedServices" vem depois do Delete ao banco, ou seja, você não está contando com o item sendo desinstalado no seu for. Beleza, então temos um mau uso de estado aqui. Por isso você sai passando essa lista de serviceNames que eu achei que era serviços não usados e na verdade são serviços usados pelo item a ser desinstalado que não está mais no banco de dados. Tá vendo o nó no raciocínio? Com toda certeza do mundo esse código não é nem de perto o seu pseudo OU o seu pseudo tá todo desorganizado. Imagino que você tenha certeza de que o código funciona, mas funcionar é o mínimo. Explicar pro próximo dev via código como funciona é o objetivo. Não está acontecendo isso, olha a dificuldade da legibilidade que seu código me causa.

Voltamos à linha 325, beleza, tu pegou os serviços em uso e em seguida:
```
if slices.Contains(servicesInUse, serviceName) {
```
		
Que serviceName é esse? Ah tá, serviceName é o iteration do installedServiceNames que eu havia me confundido. Se estamos trabalhando com DOIS tipos de service, os em uso e os installed pq raios não tem um prefixo ali? serviceName é um péssimo nome pra iteration, há DOIS tipos poxa, cadê o prefixo dele? É isso que eu falo, tá vendo como vocês tem mania de abusar da verbosidade onde não deve e esquecer totalmente dela quando deve? Na PR anterior eu falei: "se tem dois do mesmo item num contexto local (for, if, func etc) então você precisa adicionar prefixo. Falamos isso em calls várias vezes e eu já corrigi isso várias vezes. Ainda sim, EXATAMENTE o mesmo erro pois você não tá lendo o código, não tá buscando explicar sua intenção em voz alta.

Ok, voltamos ao método auxiliar, linha 354. Beleza, você tem um for pra desinstalar os serviços que não estão em uso. A mensagem de erro vai passar pra frente qual é o erro no processo de desinstalação, mas ela hora nenhuma menciona qual é o serviço sendo desinstalado. Como eu vou saber que o erro veio da desinstalação do MySQL e não do PHP por exemplo?

Outro ponto que me ocorreu. E se o usuário não quiser remover o MySQL quando ele remove o WordPress? Não damos opção pra ele pra não remover os serviços, o que é uma flag boleana que podemos add no processo de delete pra evitar todo o processo de desinstalação de serviços. Vamos adicionar?

Pronto. Isso é uma leitura em voz alta buscando entender a intenção do seu código. Tá vendo como ela detecta um monte de falha no raciocínio? Novamente, o foco tá em fazer funcionar, não está em fazer funcionar explicando a intenção e não deixando dúvida. Métodos simples, eu não deveria ter tido 1/10 das dúvidas que tive aqui.